pkg/util/util.go

```diff
@@ -126,13 +126,19 @@ func (u ChartUtils) LookupChartDir(chartDirs []string, dir string) (string, erro
 	for _, chartDir := range chartDirs {
 		currentDir := dir
 		for {
-			if FileExists(path.Join(currentDir, "Chart.yaml")) {
+			chartYaml := path.Join(currentDir, "Chart.yaml")
+			parent := path.Dir(path.Dir(chartYaml))
+
+			// check directory has a Chart.yaml and that it is in a
+			// direct subdirectory of a configured charts directory
+			if FileExists(chartYaml) && (parent == chartDir) {
```

Comment:
`parent` is always relative because `git diff --name-only` returns relative paths. `chartDir` could indeed be absolute. I just tested this and no changed directories would be detected in this case. The directory is expected to be relative to the working directory. I guess I should validate this.